pkg/storegateway/bucket.go

```diff
@@ -611,12 +611,12 @@ func (s *BucketStore) Series(req *storepb.SeriesRequest, srv storepb.Store_Serie
 
 	// Wait for the query gate only after opening blocks. Opening blocks is usually fast (~1ms),
 	// but sometimes it can take minutes if the block isn't loaded and there is a surge in queries for unloaded blocks.
-	tracing.DoWithSpan(ctx, "store_query_gate_ismyturn", func(ctx context.Context, _ tracing.Span) {
-		err = s.queryGate.Start(ctx)
-	})
+	span, spanCtx := opentracing.StartSpanFromContext(ctx, "store_query_gate_ismyturn")
+	err = s.queryGate.Start(spanCtx)
 	if err != nil {
 		return errors.Wrapf(err, "failed to wait for turn")
 	}
+	span.Finish()
```

Comment:
This will not finish span in case of error. We should move `span.Finish()` before error check.

Reply:
Oh yes, you are totally right. Well spotted and fixed in f45fb5ab7de643cdfc38746b3234634c82656990